### ISSUE TITLE
`String.encode(Encoding, :xml => :text)` double encodes ampersand

### ISSUE THREAD

**Issue by pepijnve** [oracle/truffleruby#1545]
Tested with 1.0-RC11.

This test passes on MRI. On Truffle it produces `&amp;#xB1;`.

```
describe "String.encode" do
  it 'should behave like MRI' do
    expect("\u00B1".encode(Encoding::US_ASCII, :xml => :text)).to(eq("&#xB1;"))
  end
end
```

**Comment by nirvdrum** [comment]
Thanks for the test. This looks like something that should be added to the Ruby Spec Suite. Unfortunately, `String#encode` with the `xml` option isn't completely implemented and we fail several other related specs, as seen in our tags file: https://github.com/oracle/truffleruby/blob/5975d2aa48ea24e560baa81abe62cffc89cb9b3c/spec/tags/core/string/encode_tags.txt

If you don't mind me asking, what's the use case? We need to implement it, regardless. I'm just curious because I don't think I've ever seen it used.

**Comment by pepijnve** [comment]
@nirvdrum see https://github.com/asciidoctor/asciimath/issues/10
I'm using it as a really simple way to prepare a string for embedding in other XML content.

The input string is MathML containing various mathematical symbols which are represented as their Unicode codepoint. In order to make the final MathML string as safe and simple to use as possible I opted to XML escape anything that's outside 7-bit ASCII. String.encode was the most straightforward way to achieve that.

**Comment by nirvdrum** [comment]
@pepijnve I've fixed this issue in 5667e6c2570a45133e9257d429f8e09796e3a5db and it will ship in the next TruffleRuby release. Thanks again for the detailed bug report!
